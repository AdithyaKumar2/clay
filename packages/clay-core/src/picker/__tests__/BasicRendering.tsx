/**
 * SPDX-FileCopyrightText: Â© 2022 Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: BSD-3-Clause
 */

import {cleanup, render} from '@testing-library/react';
import React from 'react';

import {Option, Picker} from '../../';

describe('Picker basic rendering', () => {
	afterEach(cleanup);

	it('render static content', () => {
		render(
			<Picker>
				<Option key="apple">Apple</Option>
				<Option key="banana">Banana</Option>
				<Option key="blueberry">Blueberry</Option>
			</Picker>
		);

		expect(document.body).toMatchSnapshot();
	});

	it('render dynamic content', () => {
		const {container} = render(
			<Picker items={['Apple', 'Banana', 'Blueberry']}>
				{(item) => <Option key={item}>{item}</Option>}
			</Picker>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders open component by default', () => {
		render(
			<Picker defaultActive items={['Apple', 'Banana', 'Blueberry']}>
				{(item) => <Option key={item}>{item}</Option>}
			</Picker>
		);

		expect(document.body).toMatchSnapshot();
	});

	it('renders component with selected by default', () => {
		const {getByRole} = render(
			<Picker
				defaultSelectedKey="Apple"
				items={['Apple', 'Banana', 'Blueberry']}
			>
				{(item) => <Option key={item}>{item}</Option>}
			</Picker>
		);

		const selectedValue = getByRole('combobox');

		expect(selectedValue.innerHTML).toBe('Apple');
	});

	it('renders component with selected by default and open', () => {
		const {getAllByRole} = render(
			<Picker
				defaultActive
				defaultSelectedKey="Apple"
				items={['Apple', 'Banana', 'Blueberry']}
			>
				{(item) => <Option key={item}>{item}</Option>}
			</Picker>
		);

		const [apple] = getAllByRole('option');

		expect(apple.getAttribute('aria-selected')).toBe('true');
		expect(apple.textContent).toBe('Apple');
	});
});
