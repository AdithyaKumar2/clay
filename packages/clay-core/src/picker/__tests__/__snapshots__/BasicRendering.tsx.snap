// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker basic rendering render dynamic content 1`] = `
<div>
  <button
    aria-activedescendant=""
    aria-expanded="false"
    aria-haspopup="listbox"
    class="form-control form-control-select form-control-select-secondary"
    role="combobox"
  >
    Select an option
  </button>
</div>
`;

exports[`Picker basic rendering render static content 1`] = `
<body>
  <div>
    <button
      aria-activedescendant=""
      aria-expanded="false"
      aria-haspopup="listbox"
      class="form-control form-control-select form-control-select-secondary"
      role="combobox"
    >
      Select an option
    </button>
  </div>
</body>
`;

exports[`Picker basic rendering renders open component by default 1`] = `
<body>
  <div>
    <button
      aria-activedescendant="Apple"
      aria-controls="clay-id-5"
      aria-expanded="true"
      aria-haspopup="listbox"
      aria-owns="clay-id-6"
      class="form-control form-control-select form-control-select-secondary show"
      role="combobox"
      style=""
    >
      Select an option
    </button>
  </div>
  <div>
    <div
      class="dropdown-menu dropdown-menu-indicator-start dropdown-menu-select show"
      id="clay-id-5"
      role="presentation"
      style="left: -999px; top: -995px;"
    >
      <ul
        class="inline-scroller list-unstyled"
        id="clay-id-6"
        role="listbox"
        tabindex="-1"
      >
        <li
          role="presentation"
        >
          <button
            aria-selected="false"
            class="dropdown-item focus"
            id="Apple"
            role="option"
            tabindex="-1"
          >
            Apple
          </button>
        </li>
        <li
          role="presentation"
        >
          <button
            aria-selected="false"
            class="dropdown-item"
            id="Banana"
            role="option"
            tabindex="-1"
          >
            Banana
          </button>
        </li>
        <li
          role="presentation"
        >
          <button
            aria-selected="false"
            class="dropdown-item"
            id="Blueberry"
            role="option"
            tabindex="-1"
          >
            Blueberry
          </button>
        </li>
      </ul>
    </div>
  </div>
</body>
`;
